plugins/outputs/otlp/otlp.go

```diff
@@ -0,0 +1,379 @@
+package otlp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sort"
+	"time"
+
+	// TODO: this should be simplified
+	// Imports the OTLP client from the prometheus sidecar
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	metricsService "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	otlpcommonpb "go.opentelemetry.io/proto/otlp/common/v1"
+	otlpmetricpb "go.opentelemetry.io/proto/otlp/metrics/v1"
+	otlpresourcepb "go.opentelemetry.io/proto/otlp/resource/v1"
+	"google.golang.org/grpc/metadata"
+)
+
+// OTLP is the OpenTelemetry Protocol config info.
+type OTLP struct {
+	Endpoint   string            `toml:"endpoint"`
+	Timeout    string            `toml:"timeout"`
```

Comment:
Please let telegraf do the heavy-lifting:
```suggestion
	Timeout    config.Duration            `toml:"timeout"`
```

Reply:
Thanks for the suggestion, this is much cleaner now!